File: 30- যোগ্য সংখ্যা ১.c
#include<stdio.h>
int main()
{

    int t;
    scanf("%d",&t);
    while(t--)
    {
        long long int m=0,n;
        scanf("%lld",&n);
        for(long long int i=1;i<n;i++)
        {
            if(n%i==0)
                m = m+i;
        }

        if(m==n)
            printf("YES, %d is a perfect number!\n",n);
        else
            printf("NO, %d is not a perfect number!\n",n);



    }
    return 0;


}

File: 30- যোগ্য সংখ্যা ১.py
t = int(input())

for i in range(t):
    x = int(input())

    p = [6,28,496,8128,33550336,8589869056,137438691328,2305843008139952128]

    if x in p:
        print("YES, {} is a perfect number!".format(x))
    else:
        print("NO, {} is not a perfect number!".format(x))
    
